add documentation structure and guidelines (#4)
ansible-community/onboarding: docs/docs_toolkit.md

> # Documenting your project

Reviewer: ```suggestion
# Documenting Ansible projects
```

This section helps you add documentation to your Ansible project or improve your existing documentation.
The onboarding [project-template](https://github.com/ansible-community/project-template)

> in GitHub gives you the scaffolding files to start your documentation effort. 

Reviewer: ```suggestion
provides a set of base files to quickly create a new documentation set. 
```

> ## Creating new documentation projects

Reviewer: ```suggestion
## Creating new documentation sets
```

Reviewer: I think we should say either "documentation set" or maybe "docsite" here. For me "documentation project" sounds like a standalone thing dedicated only to docs.

> Ansible projects starting a new documentation effort should follow these guidelines:

Reviewer: ```suggestion
Documentation guidelines for new Ansible projects:
```

* Markdown is simple, ubiquitous, and familiar to most contributors. For these reasons
 the Ansible community team recommends markdown as the format for documentation.

> * [mkdocs](https://www.mkdocs.org/getting-started/) is the recommended tool for building 
> the documentation from markdown.

Reviewer: ```suggestion
* [mkdocs](https://www.mkdocs.org/getting-started/) is the recommended tool for building 
the doc site from markdown source.
```

* [ReadtheDocs](https://readthedocs.org/) is the recommended place to publish documentation
 for Ansible projects.
* The [Ansible style guide](https://docs.ansible.com/ansible/latest/dev_guide/style_guide/index.html) 
is available to help ensure your project is well-written and follows technical documentation guidelines.

> Follow these steps to set up your documentation if you do not use the GitHub project-template:

Reviewer: ```suggestion
If you do not use the project-template, follow these steps to set up your documentation:
```

> 1. Create a ``/docs`` directory in your project repository. 

Reviewer: ```suggestion
1. Create a ``/docs`` directory at the base of your project repository. 
```

> 2. Follow the proposed [documentation structure](https://github.com/ansible-community/project-template/tree/main/docs)
>  from the project-template to ensure you include all the important guides common
> to all projects.

Reviewer: ```suggestion
2. Create the proposed [documentation structure](https://github.com/ansible-community/project-template/tree/main/docs) from the project-template to include the guides that are common to Ansible projects.
```

> 3. Add a [`mkdocs.yml`](https://github.com/ansible-community/project-template/blob/main/mkdocs.yml) to
>  your project root directory and modify the `nav` section to match your proposed navigation/set of guides.

Reviewer: ```suggestion
3. Set up your documentation build in the [`mkdocs.yml`](https://github.com/ansible-community/project-template/blob/main/mkdocs.yml) configuration file so the `nav` section includes all your project guides.
```

> 4. Create [``.readthedocs.yml``](https://github.com/ansible-community/project-template/blob/main/.readthedocs.yaml) in your project root directory.

Reviewer: ```suggestion
4. Add the Read The Docs configuration file, [``.readthedocs.yml``](https://github.com/ansible-community/project-template/blob/main/.readthedocs.yaml), to your project's base.
```

> 5. Coordinate with the [Ansible community team](https://docs.ansible.com/ansible/latest/community/communication.html#forum) to set up automated publishing and to have your documentation added to the Ansible ecosystem page.

Reviewer: ```suggestion
5. Ask the [Ansible community team](https://docs.ansible.com/ansible/latest/community/communication.html#forum) to help set up server-side configuration on Read The Docs and add your project to the Ansible namespace.
```

Reviewer: This last bit seems a little risky to me. Does this onboarding apply only to projects in the Ansible org?

Author: This is meant for 'official' ecosystem projects only. Not sure what namespace that is? While anyone can use these guidelines for any project they want, it is meant for projects that will be part of the official Ansible community projects.
